File: backend/app/models/job.py
import uuid
from sqlalchemy import Column, String, Text, DateTime, Boolean, Numeric, func
from app.core.db import Base
from app.core.db_types import GUID, JSONField


class Job(Base):
    __tablename__ = "jobs"

    id = Column(GUID(), primary_key=True, default=uuid.uuid4)
    source = Column(String(50), nullable=True)
    external_id = Column(String(200), unique=True, nullable=False)
    title = Column(String(300), nullable=False)
    company = Column(String(200), nullable=False)
    location = Column(String(200), nullable=True)
    description = Column(Text, nullable=True)
    required_skills = Column(JSONField, nullable=True)
    experience_required = Column(Numeric(4, 1), nullable=True)
    role_type = Column(String(50), nullable=True)
    domain = Column(String(100), nullable=True)
    posted_at = Column(DateTime(timezone=True), nullable=True)
    scraped_at = Column(DateTime(timezone=True), server_default=func.now())
    is_active = Column(Boolean, nullable=False, default=True)

File: frontend/src/types/job.ts
export type Job = {
  id: string;
  source?: string | null;
  external_id: string;
  title: string;
  company: string;
  location?: string | null;
  description?: string | null;
  required_skills?: unknown;
  experience_required?: number | null;
  role_type?: string | null;
  domain?: string | null;
  posted_at?: string | null;
  scraped_at: string;
  is_active: boolean;
};
